pkg/addon/helper.go

```diff
@@ -243,11 +278,104 @@ func GetObservabilityAccessibilityInfo(ctx context.Context, k8sClient client.Cli
 	return domains, nil
 }
 
+// FindWholeAddonPackagesFromRegistry find addons' WholeInstallPackage from registries, empty registryName indicates matching all
+func FindWholeAddonPackagesFromRegistry(ctx context.Context, k8sClient client.Client, addonNames []string, registryNames []string) ([]*WholeAddonPackage, error) {
+	var addons []*WholeAddonPackage
+	var registries []Registry
+
+	if len(addonNames) == 0 {
+		return nil, fmt.Errorf("no addon name specified")
+	}
+
+	registryDataStore := NewRegistryDataStore(k8sClient)
+
+	// Find matched registries
+	if len(registryNames) == 0 {
+		// Empty registryNames will match all registries
+		regs, err := registryDataStore.ListRegistries(ctx)
+		if err != nil {
+			return nil, err
+		}
+		registries = regs
+	} else {
+		// Only match specified registries
+		for _, registryName := range registryNames {
+			r, err := registryDataStore.GetRegistry(ctx, registryName)
+			if err != nil {
+				return nil, err
+			}
+			registries = append(registries, r)
+		}
+	}
+
+	if len(registries) == 0 && len(registryNames) > 0 {
+		return nil, ErrRegistryNotExist
+	}
+
+	// Found addons, for deduplication purposes
+	foundAddons := make(map[string]bool)
+	merge := func(addon *WholeAddonPackage) {
+		if _, ok := foundAddons[addon.Name]; !ok {
+			foundAddons[addon.Name] = true
+		}
+		addons = append(addons, addon)
+	}
+
+	// Find matched addons in registries
+	for _, r := range registries {
+		if IsVersionRegistry(r) {
+			vr := BuildVersionedRegistry(r.Name, r.Helm.URL, &common.HTTPOption{Username: r.Helm.Username, Password: r.Helm.Password})
+			for _, addonName := range addonNames {
+				wholePackage, err := vr.GetAddonWholePackage(ctx, addonName, "")
+				if err != nil {
+					continue
+				}
+				merge(wholePackage)
+			}
+		} else {
+			meta, err := r.ListAddonMeta()
```

Comment:
will it use cache here? I'm not sure if the performance will be very bad here. \cc @wangyikewxgm 

Maybe, we can downgrade the experience to the old mode if the registry is not versioned.

Reply:
it is only called when 
1. cli 
2. verbose is on
so it think it is not a big deal